### ISSUE TITLE
Should use mne/CUDA for resampling

### ISSUE THREAD

**Issue by drammock** [LABSN/expyfun#248]

**Comment by rkmaddox** [comment]
I'm not sure it should. It's a whole new dependency for one piece of functionality. We should at least confirm it's faster before we add the dependency.

**Comment by Eric89GXL** [comment]
All hail the mighty CUDA:
```
import numpy as np
from mne.filter import resample
from time import time
import matplotlib.pyplot as plt

durs = (0.1, 0.5, 1., 2., 5., 10.)  # sec
n_durs = len(durs)
fs_in = 44100
fs_out = 24414
n_rep = 10
n_sig = 12
t = np.empty((3, n_durs))
kinds = (1, 6, 'cuda')
for ii, dur in enumerate(durs):
    n_samp = int(dur * fs_in)
    sig = np.empty((n_sig, n_samp))
    for ti, n_jobs in enumerate(kinds):
        resample(sig, 1, 10, n_jobs=n_jobs)
        t0 = time()
        for _ in range(n_rep):
            resample(sig, 24414, 44100, n_jobs=n_jobs)
        t[ti, ii] = (time() - t0) / n_rep

colors = ['k', (0.5, 0.5, 0.5), (0., 1., 0.)]
fig, ax = plt.subplots(1, 1)
for ki, (kind, color) in enumerate(zip(kinds, colors)):
    ax.plot(durs, t[ki], color=color, label=kind)
ax.set(xscale='log', yscale='log', title='mne-python resample()')
ax.set(xlabel='Signal duration (sec)', ylabel='Processing time (sec)')
ax.set(xticks=durs, xticklabels=durs)
plt.legend(title='n_jobs=', loc='SouthEast')
plt.show()
```
Note this is a log-log plot:
![figure_1](https://cloud.githubusercontent.com/assets/2365790/9531674/3b163c50-4cd5-11e5-828c-c4059793878f.png)

**Comment by rkmaddox** [comment]
Oh my. So: require mne-python, or copy it's resample code into our own library?

**Comment by Eric89GXL** [comment]
I think requiring mne-python is okay, it's a pure python dependency anyway, even installable using `pip`.

**Comment by rkmaddox** [comment]
K. I'll do it.

**Comment by drammock** [comment]
thx for volunteering @rkmaddox

**Comment by drammock** [comment]
closed by #249

**Comment by rkmaddox** [comment]
Now that `expyfun` is using CUDA, resmampling of the CRM corpus went from 28 minutes to 1 minute. I am satisfied with that improvement.
